src/backend/executor/execUtils.c

```diff
@@ -2163,8 +2163,14 @@ void mppExecutorFinishup(QueryDesc *queryDesc)
 		int primaryWriterSliceIndex = PrimaryWriterSliceIndex(estate);
 		//if (sliceRunsOnQE(currentSlice))
 		{
-			estate->es_processed +=
-				cdbdisp_sumCmdTuples(pr, primaryWriterSliceIndex);
+			/*
+			 * Do not modify es_processed if it is not zero. Because
+			 * es_processed save info about tuples, which were really processed,
+			 * and limit was used (not to process tuples more than limit).
```

Comment:
Is there a check for this?

Reply:
Yes, it checks by block with comment: `Check, that INSERT produce the error with correct explanation`